build-support/bin/pre-commit.sh

```diff
@@ -51,36 +51,36 @@ printf "%s\n" "${ADDED_FILES[@]}" \
 echo "* Checking for banned imports"
 ./build-support/bin/check_banned_imports.sh
 
-if git diff master --name-only | grep '\.rs$' > /dev/null; then
-  echo "* Checking formatting of rust files" && ./build-support/bin/check_rust_formatting.sh || exit 1
-  # Clippy happens on a different shard because of separate caching concerns.
-  if [[ "${RUNNING_VIA_TRAVIS_CI_SCRIPT}" != "1" ]]; then
-    echo "* Running cargo clippy" && ./build-support/bin/check_clippy.sh || exit 1
-  fi
-  echo "* Checking rust target headers" && build-support/bin/check_rust_target_headers.sh || exit 1
-fi
-
 echo "* Checking for bad shell patterns" && ./build-support/bin/check_shell.sh || exit 1
 
-$(git rev-parse --verify master > /dev/null 2>&1)
-if [[ $? -eq 0 ]]; then
+# When travis builds a tag, it does so in a shallow clone without master fetched, which
+# fails in pants changed.
+if git rev-parse --verify "master" &>/dev/null; then
```

Comment:
```suggestion
# Test if the "master" branch exists in the current git checkout.
if git rev-parse --verify "master" &>/dev/null; then
```

Reply:
In order to unbreak the branch without another CI run, I'm going to land as is. Thank you for your attention to detail here though.